app/lib/primer/tabbed_component_helper.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require "active_support/concern"
+
+module Primer
+  # Helper to share tab validation logic between components.
+  # The component will raise an error if there are 0 or 2+ selected tabs.
+  module TabbedComponentHelper
+    extend ActiveSupport::Concern
+
+    included do
+      include ViewComponent::SlotableV2
+    end
+
+    class MultipleSelectedTabsError < StandardError; end
+    class NoSelectedTabsError < StandardError; end
+
+    def before_render
+      validate_single_selected_tab
+    end
+
+    private
+
+    def wrapper
+      return yield unless @with_panel
+
+      render Primer::TabContainerComponent.new do
+        yield
+      end
+    end
+
+    def validate_single_selected_tab
+      raise MultipleSelectedTabsError, "only one tab can be selected" if selected_tabs_count > 1
+      raise NoSelectedTabsError, "a tab must be selected" if selected_tabs_count != 1
+    end
```

Comment:
How do you feel about raising this error these days? I know we've gone back and forth about the risk of runtime exceptions in production.

Reply:
Yeah, this is something we need to decide sooner than later.
Maybe what we need is a `validation` API that we make sure to always skip in production

Reply:
Yeah, like `fetch_or_fallback` 😄

Reply:
https://github.com/primer/view_components/issues/395